Panic: 'file not in program' error on correct TypeScript configuration

Since updating to oxlint-tsgolint version 0.1.0, I've encountered a consistent panic error:

```
panic: file not in program: <a file indeed in the program>.
```

This error occurs despite the file being correctly included and configured within my TypeScript project.

The issue does not appear in version 0.0.4, which works as expected on the same codebase.

## Environment:

- Operating System: Windows 11
- Node.js Version: 24.6.0
- oxlint-tsgolint Version: 0.1.0, 0.1.1
- oxlint Version: 1.13.0
- TypeScript Version: 5.9.2

--- a/internal/linter/linter.go
+++ b/internal/linter/linter.go
@@ -2,7 +2,9 @@ package linter
 
 import (
 	"context"
+	"fmt"
 	"log"
+	"strings"
 
 	"github.com/typescript-eslint/tsgolint/internal/rule"
 	"github.com/typescript-eslint/tsgolint/internal/utils"
@@ -59,8 +61,21 @@ func RunLinter(logLevel utils.LogLevel, currentDirectory string, workload Worklo
 			}
 		}
 
-		for f := range fileSet {
-			panic("file not in program: " + f)
+		if len(fileSet) > 0 {
+			var unmatchedFiles []string
+			for k := range fileSet {
+				unmatchedFiles = append(unmatchedFiles, k)
+			}
+			unmatchedFilesString := strings.Join(unmatchedFiles, ", ")
+			log.Println("Unmatched files found:", unmatchedFilesString)
+
+			var programFiles []string
+			for _, k := range program.SourceFiles() {
+				programFiles = append(programFiles, k.FileName())
+			}
+			log.Printf("Program source files (%d): %s", len(programFiles), strings.Join(programFiles, ", "))
+
+			panic(fmt.Sprintf("Expected file '%s' to be in program '%s'", unmatchedFilesString, configFileName))
 		}
 
 		err = RunLinterOnProgram(program, sourceFiles, workers, getRulesForFile, onDiagnostic)

